src/client/Microsoft.Identity.Client/Internal/Requests/SilentRequest.cs

```diff
@@ -28,6 +29,7 @@ internal class SilentRequest : RequestBase
             : base(serviceBundle, authenticationRequestParameters, silentParameters)
         {
             _silentParameters = silentParameters;
+            _broker = new Lazy<IBroker>(() => ServiceBundle.PlatformProxy.CreateBroker(null));
```

Comment:
null here indicates that the broker will run in the main context rather than create a NullBroker

Reply:
i'm not sure about making a broker object for each silent call, esp when it's only used for android. let me think about this.

---
In reply to: [419790230](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1802#discussion_r419790230) [](ancestors = 419790230)

Reply:
That will not be the case, this will only be created on android and only when the broker is enabled with the WithBroker(true).

This is only initialized when all three checks are made in the conditionals that use it.

example

"ServiceBundle.PlatformProxy.CanBrokerSupportSilentAuth() &&
                AuthenticationRequestParameters.IsBrokerConfigured &&
                **_broker**.Value.CanInvokeBroker()"

Reply:
in that case I believe you should be getting the broker via the service bundle? or an additional param on the silentrequest?

Reply:
Or can you check in the CanBrokerSupportSilentAuth() method to see if it can be invoked as well? because doesn't matter if it supports it and it can't be invoked right?

---
In reply to: [419799949](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1802#discussion_r419799949) [](ancestors = 419799949)

Reply:
I meant the actual _broker object. I don't think we should create that with each silent call. 

---
In reply to: [419798837](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1802#discussion_r419798837) [](ancestors = 419798837)

Reply:
Yes, that is what i am referring to. since it is a lazy object. it will only be created when it is used on silent broker requests, which can only happen on android when the broker is enabled

Reply:
or ServiceBundle.PlatformProxy.CanInvokedBroker()

Reply:
> or ServiceBundle.PlatformProxy.CanInvokeBroker()

@henrik-me this is what I am looking at now

Reply:
Or this:
```csharp
 public override bool CanBrokerSupportSilentAuth()
        {
            IBroker broker = CreateBroker(null);
            if (broker.CanInvokeBroker())
            {
                return true;
            }
            else
            {
                return false;
            }
        }
```

---
In reply to: [419805856](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1802#discussion_r419805856) [](ancestors = 419805856)

Reply:
then in ClientApplicationBase...we might have the same issue if broker is not available...idk.
```csharp
 public async Task<IEnumerable<IAccount>> GetAccountsAsync()
        {
            RequestContext requestContext = CreateRequestContext(Guid.NewGuid());
            IEnumerable<IAccount> accounts = Enumerable.Empty<IAccount>();

            if (AppConfig.IsBrokerEnabled && ServiceBundle.PlatformProxy.CanBrokerSupportSilentAuth())
            {
                var broker = ServiceBundle.PlatformProxy.CreateBroker(null);
                if (broker.CanInvokeBroker())
                {
                    accounts = await broker.GetAccountsAsync(AppConfig.ClientId).ConfigureAwait(false);
                    return accounts;
                }
            }
...
```
so if you check in CanBrokerSupportSilentAuth that broker is also on the device, you won't have to worry about that here as well.

---
In reply to: [419806308](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1802#discussion_r419806308) [](ancestors = 419806308,419805856)

Reply:
No, the client application base is doing the same thing the silentReqeust is doing. that is where I made the fix initially.

```CSharp
var broker = ServiceBundle.PlatformProxy.CreateBroker(null);
                if (broker.CanInvokeBroker())
```

Reply:
As you can see in the ClientApplicationBase, "ServiceBundle.PlatformProxy.CreateBroker(null);" will only be called on Android and when it is enabled via WithBroker()

Reply:
yes i understand. just saying that you can't support silent auth w/out having a broker on the device, so checking in the support silent auth method makes sense to me. idk. i'm just not a fan of the broker as part of the silent request constructor.

---
In reply to: [419808536](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1802#discussion_r419808536) [](ancestors = 419808536)

Reply:
I agree that checking in the CanBrokerSupportSilentAuth() method is a good idea. reworking the code now.
@jennyf19

Reply:
ok...i think bogdan will probably have a better idea though. i just would like to move any intialization out of the constructor is all. 


---
In reply to: [419811605](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1802#discussion_r419811605) [](ancestors = 419811605)

Reply:
he might, but I think it is a good starting point atleast

Reply:
There really isnt a good way of getting around the creation of the new broker object to check if the broker is installed unless we store a static somewhere which I would prefer not to do. putting this logic in the CanBrokerSupportSilentAuth() is a good way to mitigate this issue by ensuring this will only happen on android and when the broker is enabled by the customer. I think this is a good solution with the current design we have.